tests/test_ell1h.py

```diff
@@ -12,111 +13,202 @@
+                % ("d_delay_d_" + p, np.nanmax(relative_diff).value)
+            )
+            assert np.nanmax(relative_diff) < tol, msg
+        else:
+            continue
+
+
+def test_J0613_fit_with_H3_H4(toasJ0613, modelJ0613):
+    log = logging.getLogger("TestJ0613.fit_tests")
+    f = ff.GLSFitter(toasJ0613, modelJ0613)
+    f.fit_toas()
+
+
+def test_J0613_STIG(toasJ0613, modelJ0613_STIG):
+    log = logging.getLogger("TestJ0613.fit_tests_stig")
+    f = ff.GLSFitter(toasJ0613, modelJ0613_STIG)
+    f.fit_toas()
+
+
+def test_no_H3_H4(toasJ0613):
+    """ Test no H3 and H4 in model.
+    """
+    no_H3_H4 = simple_par.replace("H4 2.0262048E-7  1       1.1276173E-7", "")
+    no_H3_H4 = no_H3_H4.replace("H3 2.7507208E-7  1       1.5114416E-7", "")
+    no_H3_H4_model = model.get_model(StringIO(no_H3_H4))
+    assert no_H3_H4_model.H3.value == None
+    assert no_H3_H4_model.H4.value == None
+    test_toas = toasJ0613[::20]
+    f = ff.WLSFitter(test_toas, no_H3_H4_model)
+    f.fit_toas()
+
+
+def test_H3_and_H4_non_zero(toasJ0613):
+    """ Testing if the different H3, H4 combination breaks the fitting. the
+        fitting result will not be checked here.
+    """
+    simple_model = model.get_model(StringIO(simple_par))
+
+    test_toas = toasJ0613[::20]
+    f = ff.WLSFitter(test_toas, simple_model)
+    f.fit_toas()
+
+
+def test_zero_H4(toasJ0613):
+    H4_zero_model = model.get_model(StringIO(simple_par))
+    H4_zero_model.H4.value = 0.0
+    H4_zero_model.H4.frozen = False
+    assert H4_zero_model.H3.value != 0.0
+    H4_zero_model.H3.frozen = False
+    test_toas = toasJ0613[::20]
+    f = ff.WLSFitter(test_toas, H4_zero_model)
+    f.fit_toas()
+
+
+def test_zero_H3(toasJ0613):
+    H3_zero_model = model.get_model(StringIO(simple_par))
+    H3_zero_model.H3.value = 0.0
+    H3_zero_model.H3.frozen = False
+    assert H3_zero_model.H4.value != 0.0
+    H3_zero_model.H4.frozen = False
+    test_toas = toasJ0613[::20]
+    with pytest.raises(ValueError):
+        f = ff.WLSFitter(test_toas, H3_zero_model)
+
+
+def test_zero_H3_H4_fit_H3_H4(toasJ0613):
+    H4_zero = simple_par.replace("H4 2.0262048E-7  1       1.1276173E-7", "H4 0  1  0")
+    H3H4_zero = H4_zero.replace("H3 2.7507208E-7  1       1.5114416E-7", "H3 0  1  0")
+    H3H4_zero_model = model.get_model(StringIO(simple_par))
+    H3H4_zero_model.H3.value = 0.0
+    H3H4_zero_model.H4.value = 0.0
+    H3H4_zero_model.H3.frozen = False
+    H3H4_zero_model.H4.frozen = False
+    test_toas = toasJ0613[::20]
+    f = ff.WLSFitter(test_toas, H3H4_zero_model)
+    f.fit_toas()
+
+
+def test_zero_H3_H4_fit_H3(toasJ0613):
+    H3H4_zero2_model = model.get_model(StringIO(simple_par))
+    H3H4_zero2_model.H3.value = 0.0
+    H3H4_zero2_model.H3.frozen = False
+    H3H4_zero2_model.H4.value = 0.0
+    H3H4_zero2_model.H4.frozen = True
+    test_toas = toasJ0613[::20]
+    f = ff.WLSFitter(test_toas, H3H4_zero2_model)
+    # This should work
+    f.fit_toas()
```

Comment:
Does fitting for H3 actually work? If all the derivatives with respect to H3 are zero (looks like they might be?) it might look like a completely degenerate parameter and the fitter would not change it at all, so fitting from this starting point would never work even if it didn't raise an exception.

Reply:
Added a derivative test that d_delay_d_H3 > 0 for that set of TOAs.

Reply:
Er, could you point to the code? The only derivative test I see is the one above that checks the claimed derivative against a numerical derivative for the special case of H4 and STIGMA both zero.